client/packages/system/src/Patient/PatientView/Footer.tsx

```diff
@@ -70,7 +70,11 @@ export const Footer: FC<FooterProps> = ({
               color="secondary"
               disabled={!isDirty || !!validationError}
               isLoading={isSaving}
-              onClick={() => showSaveConfirmation()}
+              onClick={() => {
```

Comment:
hm. tricky. 
[This is my suggestion](https://github.com/openmsupply/open-msupply/compare/1884-Patient-creation-input-data-loss-when-searching-for-a-patient-to-quickly...1884-debounce-suggestion)

The problem is that the JSON form inputs are debounced with a value of 500ms (in [this file](https://github.com/openmsupply/open-msupply/blob/0d3fb662f15ec6c8bed4e87696b34f67fae97787/client/packages/programs/src/JsonForms/common/hooks/useDebouncedTextInput.ts#L49)) and so we need to debounce the search action by at least this. There's no way of knowing that text has been entered, so we simply have to wait before searching, just in case.

It makes the modal feel slightly laggy, but I think it's the best we can do.

To test - I've copied a first name, click on the first name field in the patient search modal and then hover the cursor over the OK & Next button. Quickly paste and click the button.. without the debounce, the first name is not included in the search.

Reply:
🤔 the last name is still missing for me with that suggestion... you can enter first and last name, then click next, then enter on keyboard straight away 🤔 
will look into it some more (:

Reply:
I've removed the handleKeyDown and that seemed to solve the missing input but have debounced okNext anyway just in case (: 
[142145d](https://github.com/openmsupply/open-msupply/pull/1898/commits/142145d5fb97da1bfaa05d38881d82d952c0b969)